setuptools/dist.py

```diff
@@ -100,8 +119,9 @@ def _read_list(name):
     else:
         self.download_url = None
 
-    self.long_description = _read_field('description')
-    self.description = _read_field('summary')
+    self.long_description = _read_long_description()
+    if self.long_description is None and self.metadata_version >= StrictVersion('2.1'):
```

Comment:
Is there a solution that doesn't select on metadata version? Note that long description in Metadata 2.1 could still use the Description field.

Reply:
It will try to read the long description from the header field first. Only if the result is `None` and the `mv >= '2.1'` we try to read it from the payload as well. It will work that way, since in `2.1` an `UNKNOWN` value for the long description is added either in the header field or the payload.